### PR TITLE
Add some more tests for git_futils_rmdir_r and some cleanup

### PR DESCRIPTION
**Pull request by csware**
**Update**: I cannot reproduce the issue any more which was the initial reason for this MR, however, this is a good extension for the test cases now...

**old**: When working on another issue I found that git_futils_rmdir_r does not delete an empty path hierarchy, just the last folder

### CODE REVIEW

**Comment by pks-t** [comment]
I've fixed your tests and also added a cleanup function that always removes left-over bits of the test directory hierarchy. With these changes, all tests pass just fine, and it seems to me like the empty directory hierarchy is being deleted just fine. Also, the following simple tests passes (`__initialize` creates an empty directory hierarchy by default):

```
void test_core_rmdir__can_remove_empty_dirs(void)
{
	cl_git_pass(git_futils_rmdir_r(empty_tmp_dir, NULL, GIT_RMDIR_SKIP_NONEMPTY));
	cl_assert(!git_path_exists(empty_tmp_dir));
}
```

**Comment by csware** [comment]
I cannot reproduce the issue any more which was the initial reason for this MR, however, this is a good extension for the test cases now...

File: tests/core/rmdir.c [hunk 1]
```diff
@@ -27,10 +27,24 @@ void test_core_rmdir__initialize(void)
 	git_buf_dispose(&path);
 }
 
+void test_core_rmdir__cleanup(void)
+{
+	if (git_path_exists(empty_tmp_dir))
+		cl_git_pass(git_futils_rmdir_r(empty_tmp_dir, NULL, GIT_RMDIR_REMOVE_FILES));
```
[the remark this answers is not in the document]
Comment (pks-t) [reply tests/core/rmdir.c]: Not if a test has been written that deletes the directory itself

**Comment by pks-t** [comment]
Agreed, @csware, we should just merge this even though it doesn't catch any errors right now.

File: tests/core/rmdir.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (ethomson) [reply tests/core/rmdir.c]: But that's not clar's lifecycle - for every test, the `initialize` will run immediately before, and the `cleanup` will run immediately after.  So `core::rmdir::can_skip_non_empty_dir` can do whatever it wants, the `initialize` will recreate that directory.

If that directory doesn't exist, then something strange has happened in the test execution, and we should error so that we can investigate, IMHO.

[the remark this answers is not in the document]
Comment (pks-t) [reply tests/core/rmdir.c]: The current flow is:

1. The `__initialize` function creates `empty_tmp_dir` and some more directories inside of it
2. The test performs whatever it wants with and within empty_tmp_dir. At least one test will use `git_futils_rmdir_r` to remove `empty_tmp_dir` altogether
3. The `__cleanup` function will now try to remove `empty_tmp_dir` again. But as the test has already deleted it, it will fail

So it's wasn't anything strange, but just the test verifying that it is able to remove `empty_tmp_dir` with the function that is currently the system under test

[the remark this answers is not in the document]
Comment (ethomson) [reply tests/core/rmdir.c]: Oh, goodness.  I'm very sorry, I thought that this was another _test_ being added, I hadn't realized that it was the cleanup function itself.  😥  Thanks for reminding me.

**Comment by ethomson** [comment]
Agreed here, too.  Thanks for the PR!
